### PR TITLE
DEVEX-2226 File (un)archive support in DXJava

### PR DESCRIPTION
**Pull request by wormsik**
**Java binding changes:**
* new field in `dx describe`
* possibility to search using `archivalState` field
* possibility to (un)archive files
* suitable tests
* minor fixes in javadoc strings to be compatible with Java 11 and HTML5

**Other changes:**
* make `build/run_java_integration_tests.py` compatible with Python 3

### CODE REVIEW

File: src/java/src/main/java/com/dnanexus/DXProject.java [hunk 1]
```diff
@@ -219,6 +223,458 @@ public void destroy(boolean terminateJobs) {
...
+     * </pre>
+     *
+     * @return a newly initialized {@code ArchiveRequestBuilder}
+     */
+    public ArchiveRequestBuilder archive() {
+        return new ArchiveRequestBuilder();
+    }
+
+    /**
+     * Retrieve specified files or folder from an archive storage.
```
Comment (kpjensen) [inline src/java/src/main/java/com/dnanexus/DXProject.java:249]: Can you reword this to "Request specified files or folder to be unarchived"

File: src/java/src/main/java/com/dnanexus/DXProject.java [hunk 2]
```diff
@@ -219,6 +223,458 @@ public void destroy(boolean terminateJobs) {
                 RetryStrategy.SAFE_TO_RETRY);
     }
 
+    /**
+     * Move specified files or folder to an archive storage to save on storage costs.
```
Comment (kpjensen) [inline src/java/src/main/java/com/dnanexus/DXProject.java:227]: Can you reword this to "Request specified files or folder to be archived"

File: src/java/src/test/java/com/dnanexus/DXProjectTest.java [hunk 3]
```diff
@@ -49,6 +54,40 @@ public void tearDown() {
         }
     }
 
+    private DXFile createMinimalFile(String name) {
```
Comment (kpjensen) [inline src/java/src/test/java/com/dnanexus/DXProjectTest.java:57]: Can you rename this fn to `uploadSmallFile()` as `create` implies only calling `file-xxxx/new`.

**Review by kpjensen — APPROVED** [review]
LGTM. Minor comments to change a few comments and function names.

File: src/java/src/main/java/com/dnanexus/DXProject.java (hunk 1 above)
Comment (wormsik) [reply src/java/src/main/java/com/dnanexus/DXProject.java]: done

File: src/java/src/main/java/com/dnanexus/DXProject.java (hunk 2 above)
Comment (wormsik) [reply src/java/src/main/java/com/dnanexus/DXProject.java]: done
